### PR TITLE
Fix am / pm i18n's bug

### LINKED ISSUE #22924: AM/PM button on calendar does not activate in Japanese environment
**Describe the bug**
 AM/PM button on calendar does not activate in Japanese environment. 
datetime is updated successfully.

It is correctly reversed in English environment.

**To reproduce**
Kapture 2020-06-05 at 15 26 30

**Expected behavior**
I want the appropriate AM/PM button to be active.

**Editor version (please complete the following information):**
I tried it in the following environment.

- WordPress 5.4.1
- WordPress 5.4.1 + Gutenberg 8.2.1

**Additional context**

I think there is a problem with the following part.

https://github.com/WordPress/gutenberg/blob/c0e4f538ad2ed96eed064b39657b35f5645238a1/packages/components/src/date-time/time.js#L91

On Japanese environment, `moment().format('A');` return `午後` or `午前`.

But below is a comparison with "AM" or "PM".

https://github.com/WordPress/gutenberg/blob/c0e4f538ad2ed96eed064b39657b35f5645238a1/packages/components/src/date-time/time.js#L303-L320

### PR DESCRIPTION
**Pull request by miminari**
## Description
Fixes #22924
I changed the state `am` value to just string 'AM' and 'PM' for i18n.

## How has this been tested?
1. Make the setting of time to `g:i A`, and the language to Japanese.
2. Create a new post.
3. Document > Publish
4. You can change AM='午前', PM='午後' and that's current color.

## Screenshots
Kapture 2020-06-06 at 16 12 23

## Types of changes
Bugfix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.

### CODE REVIEW

**Comment by torounit** [comment]
@miminari 

It's helpful.

https://github.com/moment/moment/blob/a0f3c74374b638b769d9ab1ae54ba5266ef0976f/moment.js#L1996-L2002

**Comment by miminari** [comment]
Thanks, @torounit .
I changed the way to the 'am' state value to just string 'AM' or 'PM'.

File: packages/components/src/date-time/time.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ class TimePicker extends Component {
 		const month = selected.format( 'MM' );
 		const year = selected.format( 'YYYY' );
 		const minutes = selected.format( 'mm' );
-		const am = selected.format( 'A' );
+		const am = selected.format( 'A' ) === __( 'AM' ) ? 'AM' : 'PM';
```
Comment (torounit) [inline packages/components/src/date-time/time.js:92]: I think it's better to compare it to the numbers.
If the translation in moment.js is different from the translation in WordPress, a problem occurs.

Comment (torounit) [reply packages/components/src/date-time/time.js:92]: It is better to compare it with `selected.format("H")` .

**Review by torounit — CHANGES_REQUESTED** [review]

**Comment by torounit** [comment]
@miminari 
Looks good!

**Review by torounit — APPROVED** [review]

**Comment by miminari** [comment]
@torounit Thank you for your review!

**Pull request merged**